src/Core/System.php

```diff
@@ -21,15 +21,206 @@
+	 */
+	private $basePath;
+
+	public function __construct(LoggerInterface $logger, App\Mode $mode, IManageConfigValues $config, string $basepath)
+	{
+		$this->logger   = $logger;
+		$this->mode     = $mode;
+		$this->config   = $config;
+		$this->basePath = $basepath;
+	}
+
+	/**
+	 * Checks if the maximum number of database processes is reached
+	 *
+	 * @return bool Is the limit reached?
+	 */
+	public function isMaxProcessesReached(): bool
+	{
+		// Deactivated, needs more investigating if this check really makes sense
+		return false;
+
+		/*
+		 * Commented out to suppress static analyzer issues
+		 *
+		if ($this->mode->isBackend()) {
+			$process = 'backend';
+			$max_processes = $this->config->get('system', 'max_processes_backend');
+			if (intval($max_processes) == 0) {
+				$max_processes = 5;
+			}
+		} else {
+			$process = 'frontend';
+			$max_processes = $this->config->get('system', 'max_processes_frontend');
+			if (intval($max_processes) == 0) {
+				$max_processes = 20;
+			}
+		}
+
+		$processlist = DBA::processlist();
+		if ($processlist['list'] != '') {
+			$this->logger->debug('Processcheck: Processes: ' . $processlist['amount'] . ' - Processlist: ' . $processlist['list']);
+
+			if ($processlist['amount'] > $max_processes) {
+				$this->logger->debug('Processcheck: Maximum number of processes for ' . $process . ' tasks (' . $max_processes . ') reached.');
+				return true;
+			}
+		}
+		return false;
+		 */
+	}
+
+	/**
+	 * Checks if the minimal memory is reached
+	 *
+	 * @return bool Is the memory limit reached?
+	 */
+	public function isMinMemoryReached(): bool
+	{
+		$min_memory = $this->config->get('system', 'min_memory', 0);
+		if ($min_memory == 0) {
+			return false;
+		}
+
+		if (!is_readable('/proc/meminfo')) {
+			return false;
+		}
+
+		$memdata = explode("\n", file_get_contents('/proc/meminfo'));
+
+		$meminfo = [];
+		foreach ($memdata as $line) {
+			$data = explode(':', $line);
+			if (count($data) != 2) {
+				continue;
+			}
+			[$key, $val]     = $data;
+			$meminfo[$key]   = (int)trim(str_replace('kB', '', $val));
+			$meminfo[$key]   = (int)($meminfo[$key] / 1024);
+		}
+
+		if (!isset($meminfo['MemFree'])) {
+			return false;
+		}
+
+		$free = $meminfo['MemFree'];
+
+		$reached = ($free < $min_memory);
+
+		if ($reached) {
+			$this->logger->warning('Minimal memory reached.', ['free' => $free, 'memtotal' => $meminfo['MemTotal'], 'limit' => $min_memory]);
+		}
+
+		return $reached;
+	}
+
+	/**
+	 * Checks if the maximum load is reached
+	 *
+	 * @return bool Is the load reached?
+	 */
+	public function isMaxLoadReached(): bool
+	{
+		if ($this->mode->isBackend()) {
+			$process    = 'backend';
+			$maxsysload = intval($this->config->get('system', 'maxloadavg'));
+			if ($maxsysload < 1) {
+				$maxsysload = 50;
+			}
+		} else {
+			$process    = 'frontend';
+			$maxsysload = intval($this->config->get('system', 'maxloadavg_frontend'));
+			if ($maxsysload < 1) {
+				$maxsysload = 50;
+			}
```

Comment:
We should remove this case, it doesn't makes sense to bar access to the frontend, Friendica is more demanding resource-wise than before, so we shouldn't try to accommodate anemic servers by default. Worker load limit makes more sense because the background tasks can be delayed a bit without being very noticeable like a blank page for the frontend.

Reply:
I removed all the `maxloadavg_frontend` usages, see https://github.com/friendica/friendica/pull/10941/commits/65697142f76fc2904924af744df1847a918672b5